apps/docs/src/app/platform/component-docs/platform-forms/platform-date-picker/platform-date-picker-examples/platform-date-picker-complexi18n.component.ts

```diff
@@ -0,0 +1,170 @@
+            (buttonClicked)="setLocale('de')"
+            [buttonType]="isSelected('de') ? 'emphasized' : ''"
+            >German</fdp-button
+        >
+        |
+        <fdp-button
+            [contentDensity]="'compact'"
+            (buttonClicked)="setLocale('bg')"
+            [buttonType]="isSelected('bg') ? 'emphasized' : ''"
+            >Bulgarian
+        </fdp-button>
+        |
+        <fdp-button
+            [contentDensity]="'compact'"
+            (buttonClicked)="setLocale('pl')"
+            [buttonType]="isSelected('pl') ? 'emphasized' : ''"
+            >Polish</fdp-button
+        >
+
+        <br />
+        <fdp-form-group [multiLayout]="true">
+            <fdp-form-field
+                [id]="'birthday'"
+                zone="zLeft"
+                rank="1"
+                [label]="'Date Format:'"
+                [placeholder]="'Enter date format'"
+            >
+                <fd-input-group
+                    id="docs-date-picker-format"
+                    [placeholder]="'Date Format'"
+                    [(ngModel)]="actualFormat"
+                    class="docs-example-fd-form-group"
+                    [button]="true"
+                    (addOnButtonClicked)="refresh()"
+                    [addOnText]="'Refresh'"
+                >
+                </fd-input-group>
+                <fd-form-message type="help">
+                    The full list of formats can be found at
+                    <a href="https://angular.io/api/common/DatePipe" target="_blank">
+                        Official Angular Documentation
+                    </a>
+                </fd-form-message>
+            </fdp-form-field>
+
+            <fdp-form-field
+                [id]="'datepicker'"
+                zone="zRight"
+                rank="1"
+                [label]="'Selected:'"
+                [placeholder]="'dd/MM/yyyy'"
+            >
+                <fdp-date-picker
+                    [name]="'datepicker'"
+                    [format]="actualFormat"
+                    [locale]="actualLocale"
+                    [(ngModel)]="date"
+                ></fdp-date-picker>
+            </fdp-form-field>
+        </fdp-form-group>
+    `,
+    providers: [
+        {
+            provide: CalendarI18n,
+            useClass: CustomI18nMomentCalendar
+        }
+    ]
+})
+export class PlatformDatePickerComplexI18nExampleComponent {
+    @ViewChild(DatePickerComponent) datePicker: DatePickerComponent;
+
+    actualLocale = '';
+
+    actualFormat = 'shortDate';
+
+    actualMomentJsLang = '';
+
+    public date: FdDate = FdDate.getToday();
+
+    constructor(private calendarI18nService: CalendarI18n) {
+        registerLocaleData(localeFrench, 'fr');
+        registerLocaleData(localePolish, 'pl');
+        registerLocaleData(localeBulgarian, 'bg');
+        registerLocaleData(localeGb, 'en-gb');
+        registerLocaleData(localeDe, 'de');
+        moment.locale('en-gb');
```

Comment:
why is the default `en-gb` instead of the default `en`?

Reply:
changed to 'en'